TwilioVerifyDemo/TwilioVerifyDemo/Common/AppDelegate.swift

```diff
@@ -82,28 +86,60 @@ extension AppDelegate: UNUserNotificationCenterDelegate {
       return
     }
     
-    showChallenge(payload: payload)
+    handleChallengeApproval(with: payload)
```

Comment:
Does this apply for foreground? Should it be only for userNotificationCenter(_:willPresent:withCompletionHandler:)?

Reply:
Not especially for foreground, this method get triggered when the user tap the actual notification, so if the user has `silent approval` enabled, it will work as expected: the app will open and approve the challenge, then open the challenge detail screen with the challenge updated. Let me know if this is the expected behavior, if not I can reset this.

Reply:
The notification will be shown when the app is in background, right? If this method is related to background, it would be better to only show the challenge for this case and only approve silently for the foreground method